helper/schema/resource.go

```diff
@@ -638,6 +638,7 @@ func (r *Resource) RefreshWithoutUpgrade(
 		state = nil
 	}
 
+	schemaMap(r.Schema).handleDiffSuppressOnRefresh(s, state)
```

Comment:
Aside: It may be beneficial to offer this as a `ResourceData` method on top of the schema field implementation, so provider developers can opt into setting this once per resource (or potentially across the provider, if we did some other implementation work). Not saying anything different needs to be done here now, unless we think that is something we want to consider upfront in terms of implementation and doing it outside of `schemaMap`.

Reply:
I did originally consider making this a whole-resource-type-level settable, but decided to go more surgical so that it would be less risky to enable for a long-standing existing resource type implementation.

I'm not sure if I follow exactly what you mean by a `ResourceData` method here, but perhaps my comment on the other question above addresses why it's considerably harder to get a good result for this at the `ResourceData`/`FieldWriter` layer and why I ultimately elected to make it a postprocessing step. If you're thinking of something different than what I described there then I'm happy to consider another approach!